cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -333,7 +333,7 @@ private Datastore createDatastore(String dbName, KeyProvider provider) throws Da
             // if it does not exist
 
             //Pass database directory, database name, and SQLCipher key provider
-            DatastoreImpl ds = new DatastoreImpl(dbDirectory, dbName, provider);
+            QueryableDatastore ds = new QueryableDatastore(dbDirectory, dbName, provider);
```

Comment:
I think you missed the point of using the inheritance in this case. The user would never see the back ends you have listed. The simple idea for using the inheritance model was to cut the line count down from the datastore, so we could match `CDTDatastore`'s interface to query e.g. 

```java
Datastore ds = manager.getDatastore("example");
ds.ensureIndexed(....);
ds.find(....);
```

So just to clear up, this doesn't propose you have a bunch of classes which you'd expose depending on the capabilities, you'd get a single interface `Datastore` in this case, which would have all those methods, so a user would only have a single instance which could be `LargeAttachmentDatastore` (being the last set of function we added), but they wouldn't know they'd just have the new `LargeAttachmentDatastore` methods appear.

`Façade Pattern` will give us what we want in the single unifying interface to make it appear more like `CDTDatastore` (in fact it will work more like `CDTDatastore` than it currently does). And in long term I think it if we ever add Geo capabilities façade will give us the best option. So I think that is the option to go for.